upup/models/bindata.go

```diff
@@ -33358,7 +33358,7 @@ data:
   # You must set a non-zero value for Typha replicas below.
   typha_service_name: "{{- if .Networking.Calico.TyphaReplicas -}}calico-typha{{- else -}}none{{- end -}}"
   # Configure the backend to use.
-  calico_backend: "bird"
+  calico_backend: {{ .Networking.Calico.Backend }}
```

Comment:
I wonder why exactly we need `Backend`...
```suggestion
  {{- if ne CalicoIPv4PoolIPIPMode "Never" }}
  calico_backend: "bird"
  {{- else }}
  calico_backend: "vxlan"
  {{- end }}
```

Reply:
One thing that I think we’d get wrong is that the “Always” and “Never” values only impact at most one _IPPool_ object, but it’s possible to operate Calico with several such _IPPools_, each configured differently. An operator could decide that the default _IPPool_ should use VXLAN, but another pool should use IP-in-IP always, and another pool should use IP-in-IP only across subnet boundaries.

With that arrangement, we’d set the backend to “vxlan,” which would preclude the operator from creating those other _IPPools_ like that using IP-in-IP encapsulation.

Those two “mode” variables only induce Calico to create an _IPPool_ like that if no other _IPPools_ exist. But even after that, an operator can edit that default pool, create more of them, and even delete the default _IPPool_.

Reply:
Thanks for the explanation.